### PR TITLE
Add a few simple rig commandline tools

### PR DESCRIPTION
**Pull request by mossblaser**
Add a selection of simple single-purpose tools which are useful when working with rig:

* [x] `rig-boot` -- boot a SpiNNaker system
* [x] `rig-power` -- power on/off a SpiNNaker machine (via the BMP)
* [x] `rig-info` -- Given a SpiNNaker board or BMP, report basic info:
  * [x] Software version
  * [x] Working Chips/Cores/Links (SpiNNaker only)
  * [x] Temperature/Fan-speed (BMP only)

### CODE REVIEW

File: rig/machine.py [hunk 1]
```diff
@@ -92,6 +94,11 @@ def from_vector(cls, vector):
 
         return _link_direction_lookup[(x, y)]
 
+    def to_vector(self):
+        """Given a link direction, return the equivilent vector."""
```
Comment (mundya) [inline rig/machine.py]: "equiv_a_lent"

File: rig/scripts/rig_boot.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            sys.stderr.write(
+                "{}: error: system already booted with {} v{}\n".format(
+                    parser.prog,
+                    info.version_string.split("/")[0],
+                    info.version))
+            return 1
+        else:
+            sys.stderr.write(
+                "{}: error: host is not a SpiNNaker system "
+                "({} running {} v{})\n".format(
```
Comment (mundya) [inline rig/scripts/rig_boot.py]: Is the intent here that you may be accidentally talking to a BMP?

**Comment by mundya** [comment]
`rig_info.py` is missing?

File: rig/scripts/rig_boot.py (hunk 2 above)
Comment (mossblaser) [inline rig/scripts/rig_boot.py]: Correct. I've done this and been confused when what follows is an apparently successful boot.

**Comment by mossblaser** [comment]
`rig_info.py` is missing?

How embarrassing! Fixed...

File: rig/scripts/rig_boot.py (hunk 2 above)
Comment (mundya) [inline rig/scripts/rig_boot.py]: I can imagine!

File: rig/scripts/rig_info.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+        else:
+            links_to_dead_chips += 1
+
+    yield "Dead links: {} (+ {} to dead/missing cores)".format(
+        dead_links, links_to_dead_chips)
+
+    yield ""
+
+    # Report running applications. Builds a histogram {app_name: {state:
+    # count}}
```
Comment (mundya) [inline rig/scripts/rig_info.py]: I guess this is good motivation for #35

**Comment by mundya** [comment]
LGTM! Merging.

**Pull request merged**